WrathCombo/AutoRotation/AutoRotationController.cs

```diff
@@ -967,19 +1048,22 @@ private static bool TargetHasRegen(IGameObject? target)
             if (target is null) return false;
             return JobID switch
             {
-                Job.AST => HasStatusEffect(AST.Buffs.AspectedBenefic, target, true),
-                Job.WHM => HasStatusEffect(WHM.Buffs.Regen, target, true),
+                Job.AST => HasStatusEffect(AST.Buffs.AspectedBenefic, target),
+                Job.WHM => HasStatusEffect(WHM.Buffs.Regen, target),
                 _ => false,
             };
         }
         private static bool TargetHasExcog(IGameObject? target)
+        {
+            return target is not null && HasStatusEffect(SCH.Buffs.Excogitation, target, true);
+        }
+        private static bool TargetHasImmortality(IGameObject? target)
         {
             if (target is null) return false;
-            return JobID switch
-            {
-                Job.SCH => HasStatusEffect(SCH.Buffs.Excogitation, target, true),
-                _ => false,
-            };
+            
+            return GetStatusEffectRemainingTime(DRK.Buffs.LivingDead, target, true) >= 3 ||
+                   GetStatusEffectRemainingTime(DRK.Buffs.WalkingDead, target, true) >= 5 ||
+                   GetStatusEffectRemainingTime(WAR.Buffs.Holmgang, target, true) >= 5;
```

Comment:
If we're being picky about how long the invlun needs to have left before resuming healing, why not include PLD & GNB?